### PR TITLE
New module - semodule

### PR DESCRIPTION
**Pull request by bandit145**
##### SUMMARY

Adds an semodule Ansible module for compiling and applying .te (Type enforcement files) as discussed in https://github.com/ansible/ansible/issues/27349

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

semodule

##### ADDITIONAL INFORMATION

Most of the logic for controlling this module is written in the action, this seemed like a good way to cut down on needing to copy the .te file to the target host if not needed.

```paste below
.te file:
module ansible-semodule-test 1.0;

require {
	type proc_net_t;
	type sshd_t;
	class file { open read };
}

allow sshd_t proc_net_t:file { open read};

- name: install test module
  semodule:
    src: test.te

TASK [semodule : install test module] *********************************
changed: [testhost] => {"changed": true, "name": "ansible-semodule-test", "version": "1.0"}

TASK [semodule : debug] ********************************************************
ok: [testhost] => {
    "test1_mod_install": {
        "changed": true, 
        "failed": false, 
        "name": "ansible-semodule-test", 
        "version": "1.0"
    }
}
```

### CODE REVIEW

**Comment by bandit145** [comment]
Due to investigation prompted by failing rhel8 and fedora tests it seems newer versions of semodule do not display module versions any-more, as such I will  drop "latest" support and operate only off module name info.

Relevant bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1690779

File: lib/ansible/modules/system/semodule.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: semodule
+short_description: Manage selinux policy modules
+description:
+  - Use this module to manage selinux policy modules from Type Enforcement files
+version_added: "2.9"
```
Comment (goneri) [inline lib/ansible/modules/system/semodule.py]: Hi @bandit145, You need to bump this to 2.10 now.

Comment (bandit145) [reply lib/ansible/modules/system/semodule.py]: Will do.

**Comment by bandit145** [comment]
@goneri Can you review this again? Thanks.

**Comment by bandit145** [comment]
There is an issue with the current opensuse15 integration test container, the container is 15.0 and it looks like the 15.0 SELinux repository is gone. I did try the 15.1 repo on opensuse 15.0 which did work but that would mean hard coding the 15.1 repository into the test and that would be really brittle in the future.

I'm leaning towards disabling the opensuse15 integration test but if the container is going to be updated soon I won't.

15.1 repo: https://download.opensuse.org/repositories/security:/SELinux/openSUSE_Leap_15.1/security:SELinux.repo

15.0 repo: https://download.opensuse.org/repositories/security:/SELinux/openSUSE_Leap_15.0/security:SELinux.repo

**Pull request closed without merging**